### PR TITLE
fix: directly trigger DOI Registrar Client update from publication api #NP-45622

### PR DESCRIPTION
**Pull request by torbjokv**

### CODE REVIEW

File: publication-rest/src/main/java/no/unit/nva/publication/delete/DeletePublicationHandler.java [hunk 1]
```diff
@@ -127,6 +141,12 @@ private void handleSoftDeletion(RequestInfo requestInfo, Publication publication
         var duplicate = requestInfo.getQueryParameterOpt(DUPLICATE_QUERY_PARAM).orElse(null);
         resourceService.unpublishPublication(toPublicationWithDuplicate(duplicate, publication));
         persistNotification(publication);
+        eventBridgeClient.putEvents(PutEventsRequest.builder().entries(PutEventsRequestEntry.builder()
```
Comment (monaullah) [inline publication-rest/src/main/java/no/unit/nva/publication/delete/DeletePublicationHandler.java:144]: extract to own method?

File: publication-rest/src/main/java/no/unit/nva/publication/delete/DeletePublicationHandler.java [hunk 2]
```diff
@@ -127,6 +141,12 @@ private void handleSoftDeletion(RequestInfo requestInfo, Publication publication
...
+               .source(NVA_PUBLICATION_DELETE_SOURCE)
+               .detailType(LAMBDA_DESTINATIONS_INVOCATION_RESULT_SUCCESS)
+               .detail(DoiMetadataUpdateEvent.createUpdateDoiEvent(publication).toJsonString())
+               .resources(publication.getIdentifier().toString())
+               .build()).build());
```
Comment (truhacevkir) [inline publication-rest/src/main/java/no/unit/nva/publication/delete/DeletePublicationHandler.java:149]: Should be persisted if needed, etc. when publication has nva provided doi. Or not?

Comment (truhacevkir) [reply publication-rest/src/main/java/no/unit/nva/publication/delete/DeletePublicationHandler.java:149]: I would extract this piece of code to new method:
`persistDoiUpdateEvent(publication);`

File: publication-rest/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java [hunk 3]
```diff
@@ -98,7 +107,11 @@ public void setUp() throws NotFoundException {
         prepareIdentityServiceClient();
         publicationService = new ResourceService(client, Clock.systemDefaultZone());
         ticketService = new TicketService(client);
-        handler = new DeletePublicationHandler(publicationService, ticketService, environment, identityServiceClient);
+        eventBridgeClient = mock(EventBridgeClient.class);
```
Comment (truhacevkir) [inline publication-rest/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java:110]: We have test implementation of EventBridgeClient.

`eventBridgeClient = new FakeEventBridgeClient(ApplicationConstants.EVENT_BUS_NAME);
`
You can than use something like this to verify that event has been persisted:

```
return eventBridgeClient.getRequestEntries()
                   .stream()
                   .map(PutEventsRequestEntry::detail)
                   .map(event -> 
attempt(() -> JsonUtils.dtoObjectMapper.readValue(event, YourClass.class)).orElseThrow())
                   .collect(Collectors.toList());
```

File: publication-rest/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java [hunk 4]
```diff
@@ -286,6 +301,8 @@ void shouldSuccessWhenUserIsContributorWhenUnpublishingPublication()
 
         var response = GatewayResponse.fromOutputStream(outputStream, Void.class);
         assertThat(response.getStatusCode(), is(equalTo(SC_ACCEPTED)));
+
+        verify(eventBridgeClient, Mockito.times(1)).putEvents(argThat(new EventBridgeRequestMatcher()));
```
Comment (monaullah) [inline publication-rest/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java:305]: I think I would assert this explicitly in another test: `shouldProduceUpdateDoiEventWhenDeletionIsSuccessful()`, and assert that the PutEventsRequestEntry contains the expected detail (DoiMetadataUpdateEvent)

**Review by monaullah — CHANGES_REQUESTED** [review]

**Pull request merged**